File: main/src/main/java/lavaplayer/source/soundcloud/SoundCloudDataReader.java
package lavaplayer.source.soundcloud;

import lavaplayer.tools.JsonBrowser;
import lavaplayer.track.AudioTrackInfo;

import java.util.List;

public interface SoundCloudDataReader {
    JsonBrowser findTrackData(JsonBrowser rootData);

    String readTrackId(JsonBrowser trackData);

    boolean isTrackBlocked(JsonBrowser trackData);

    AudioTrackInfo readTrackInfo(JsonBrowser trackData, String identifier);

    List<SoundCloudTrackFormat> readTrackFormats(JsonBrowser trackData);

    JsonBrowser findPlaylistData(JsonBrowser rootData);

    String readPlaylistName(JsonBrowser playlistData);

    String readPlaylistIdentifier(JsonBrowser playlistData);

    List<JsonBrowser> readPlaylistTracks(JsonBrowser playlistData);
}

File: main/src/main/java/lavaplayer/source/soundcloud/SoundCloudDataReader.kt
package lavaplayer.source.soundcloud

import lavaplayer.tools.JsonBrowser
import lavaplayer.track.AudioTrackInfo

interface SoundCloudDataReader {
    fun findTrackData(rootData: JsonBrowser): JsonBrowser

    fun readTrackId(trackData: JsonBrowser): String

    fun isTrackBlocked(trackData: JsonBrowser): Boolean

    fun readTrackInfo(trackData: JsonBrowser, identifier: String): AudioTrackInfo

    fun readTrackFormats(trackData: JsonBrowser): List<SoundCloudTrackFormat>

    fun findPlaylistData(rootData: JsonBrowser): JsonBrowser

    fun readPlaylistName(playlistData: JsonBrowser): String

    fun readPlaylistIdentifier(playlistData: JsonBrowser): String

    fun readPlaylistTracks(playlistData: JsonBrowser): List<JsonBrowser>
}
